From fbbd57f7e071cc787ca0ef6e2549457240490a37 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Thu, 3 Sep 2015 00:02:22 +0800
Subject: [PATCH 0587/5242] MLK-11491 ARM: imx: dis-connect vddhigh_in and
 vddsnvs_in for imx6ul

commit  c65cef96cbdd1ef985cb88751ac5b3b4de6e4ed6 from
https://source.codeaurora.org/external/imx/linux-imx.git

To achieve lower power number in low power idle, on the boards
which has higher vddhigh_in than vddsnvs_in, need to dis-connect
vddhigh_in and vddsnvs_in internally to make vddhigh_in's power
low enough to meet the design target.

Signed-off-by: Anson Huang <b20788@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/imx6ul_low_power_idle.S |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-imx/imx6ul_low_power_idle.S b/arch/arm/mach-imx/imx6ul_low_power_idle.S
index 5db724e..26bb83d 100644
--- a/arch/arm/mach-imx/imx6ul_low_power_idle.S
+++ b/arch/arm/mach-imx/imx6ul_low_power_idle.S
@@ -310,6 +310,11 @@
 	orr	r7, r7, #0x6000
 	str	r7, [r10, #0x150]
 
+	/* disconnect vdd_high_in and vdd_snvs_in */
+	ldr	r7, [r10, #0x150]
+	orr	r7, r7, #0x1000
+	str	r7, [r10, #0x150]
+
 	.endm
 
 	.macro anatop_exit_idle
-- 
1.7.9.5

